File: 1584-min-cost-to-connect-all-points/1584-min-cost-to-connect-all-points.js
/**
 * @param {number[][]} points
 * @return {number}
 */
var minCostConnectPoints = function(points) {
  const n = points.length;
  const dicts = new Array(n).fill(Infinity);
  let [cost, next] = [0, 0];
  dicts[0] = 1;
  
  for (let step = 1; step < n; step++) {
    let [min, pos] = [Infinity, -1];
    for (let i = 1; i < n; i++) {
      const [x1, y1] = points[i];
      const [x2, y2] = points[next];
      if (dicts[i]) {
        dicts[i] = Math.min(dicts[i], Math.abs(x1 - x2) + Math.abs(y1 - y2));
        if (dicts[i] < min) [min, pos] = [dicts[i], i];  
      }     
    }
    
    cost += min;
    dicts[pos] = 0;
    next = pos;
  }
  
  return cost;
};

File: 1584-min-cost-to-connect-all-points/1584-min-cost-to-connect-all-points.py
class Solution:
    def minCostConnectPoints(self, points: List[List[int]]) -> int:
        n = len(points)
        distances = [float('inf')] * n
        cost, next_pos = 0, 0
        distances[0] = 1
        
        for _ in range(1, n):
            min_dist, pos = float('inf'), -1
            for i in range(1, n):
                x1, y1 = points[i]
                x2, y2 = points[next_pos]
                if distances[i]:
                    distances[i] = min(distances[i], abs(x1 - x2) + abs(y1 - y2))
                    if distances[i] < min_dist:
                        min_dist, pos = distances[i], i
            
            cost += min_dist
            distances[pos] = 0
            next_pos = pos
        
        return cost